language/tools/move-unit-test/src/test_reporter.rs

```diff
@@ -219,6 +226,87 @@ impl TestFailure {
         }
     }
 
+    fn get_line_number(
+        loc: &Loc,
+        files: &SimpleFiles<Symbol, &str>,
+        file_mapping: &HashMap<FileHash, usize>,
+    ) -> String {
+        Self::get_line_number_internal(loc, files, file_mapping)
+            .unwrap_or_else(|_| "no_source_line".to_string())
+    }
+
+    fn get_line_number_internal(
+        loc: &Loc,
+        files: &SimpleFiles<Symbol, &str>,
+        file_mapping: &HashMap<FileHash, usize>,
+    ) -> std::result::Result<String, codespan_reporting::files::Error> {
+        let id = file_mapping
+            .get(&loc.file_hash())
+            .ok_or(codespan_reporting::files::Error::FileMissing)?;
+        let start_line_index = files.line_index(*id, loc.start() as usize)?;
+        let start_line_number = files.line_number(*id, start_line_index)?;
+        let end_line_index = files.line_index(*id, loc.end() as usize)?;
+        let end_line_number = files.line_number(*id, end_line_index)?;
+        if start_line_number == end_line_number {
+            Ok(start_line_number.to_string())
+        } else {
+            Ok(format!("{}-{}", start_line_number, end_line_number))
+        }
+    }
+
+    fn report_exec_state(test_plan: &TestPlan, exec_state: &ExecutionState) -> String {
```

Comment:
I'm not sure how hard it is to implement this in a way that does not depend on `TestPlan` (and definitely not needed for this PR). But if we could make this a function of `ExecutionState` + `Option<SourceMap>` (or some trait that `TestPlan` could implement), I could definitely anticipate other tooling benefiting from the friendlier stack traces (e.g., `move sandbox run` in the CLI).

Reply:
I have been staring at this code for a while and I don't quite see a nice and generic way of refactoring this, mostly because we need quite a bit of info that could be conceivably obtained in different ways, for example:
- source map and compiled module (of course)
- mapping from file hash to file name
- `SimpleFiles` instance to get info on byte locations of line starts

I am not saying that the refactoring is impossible, I just think that it should happen once we have a concrete other place where we want to reuse this functionality, rather than making it proactively.